### PR TITLE
Fut 7 total energy of product

### PR DESCRIPTION
**Pull request by karollayna**

### CODE REVIEW

File: Future/R/energy.R [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+   if (!is.numeric(fat)){
+      stop("fat must be numeric")
+   }
+   if (!is.numeric(carbohydrates)){
+      stop("carbohydrates must be numeric")
+   }
+   if (!is.numeric(fiber)){
+      stop("fiber must be numeric")
+   }
+   pr <- protein * 4
```
Comment (darsoo) [inline Future/R/energy.R:33]: please use better names

File: Future/R/energy.R [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+#'
+#' @export
+energy_of_product <- function(product, weight){
+   if (!is.character(product)){
+      stop("product must be string")
+   }
+   if (!is.numeric(weight)){
+      stop("weight must be numeric")
+   }
+   energy_table <- read.table(system.file("Zeszyt1.tsv", package = "Future"), sep = "\t", header = T)
```
Comment (darsoo) [inline Future/R/energy.R:98]: rename tsv file

File: Future/R/energy.R [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+   total_product_info <- energy_table %>% filter(Nazwa %in% product)
+   energy <- energy_total(weight_of_product = weight,
+                          protein = as.numeric(total_product_info$Białko..g.),
```
Comment (darsoo) [inline Future/R/energy.R:101]: Change colnames, don't use dots

File: Future/tests/testthat/test_energy.R [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  testthat::expect_equal(result, 84.3)
+})
+
+
+testthat::test_that(desc = "energy_value_of_the_product_per_100 has expected error messeges", {
+
+  testthat::expect_error(energy_value_of_the_product_per_100("2.7", 1.5, 16), "protein must be numeric")
+  testthat::expect_error(energy_value_of_the_product_per_100(2.7, "1.5", 16), "fat must be numeric")
+  testthat::expect_error(energy_value_of_the_product_per_100(2.7, 1.5, "16"), "carbohydrates must be numeric")
+  testthat::expect_error(energy_value_of_the_product_per_100("2.7", 1.5, 16, 2), "protein must be numeric")
```
Comment (darsoo) [inline Future/tests/testthat/test_energy.R:17]: These tests are not necessary (duplicated in 14-16). But we cen keep them

File: Future/tests/testthat/test_energy.R [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+})
+
+testthat::test_that(desc = "energy_total has expected error messeges", {
+
+  testthat::expect_error(energy_total("50", 1.2, 0.9, 11.8), "weight_of_product must be numeric")
+  testthat::expect_error(energy_total(50, "1.2", 0.9, 11.8), "protein must be numeric")
+  testthat::expect_error(energy_total(50, 1.2, "0.9", 11.8), "fat must be numeric")
+  testthat::expect_error(energy_total(50, 1.2, 0.9, "11.8"), "carbohydrates must be numeric")
+  testthat::expect_error(energy_total("50", 1.2, 0.9, 11.8, 3), "weight_of_product must be numeric")
```
Comment (darsoo) [inline Future/tests/testthat/test_energy.R:39]: as above

**Review by darsoo — APPROVED** [review]
minor changes, and good to go

File: Future/DESCRIPTION [hunk 6]
```diff
@@ -8,3 +8,7 @@ Description: Backend functions for shiny app Future.
 License: GPL-2
 Encoding: UTF-8
 LazyData: true
+Suggests:
```
Comment (darsoo) [inline Future/DESCRIPTION:11]: Please bump version in 4, and add file to track changes e.g. `NEWS.md`

Comment (karollayna) [reply Future/DESCRIPTION:11]: done

File: Future/R/energy.R (hunk 1 above)
Comment (karollayna) [reply Future/R/energy.R]: done

File: Future/R/energy.R (hunk 2 above)
Comment (karollayna) [reply Future/R/energy.R]: done

File: Future/R/energy.R (hunk 3 above)
Comment (karollayna) [reply Future/R/energy.R]: done

File: Future/tests/testthat/test_energy.R (hunk 4 above)
Comment (karollayna) [reply Future/tests/testthat/test_energy.R:17]: ok

File: Future/tests/testthat/test_energy.R (hunk 5 above)
Comment (karollayna) [reply Future/tests/testthat/test_energy.R:39]: ok

File: Future/DESCRIPTION (hunk 6 above)
Comment (karollayna) [reply Future/DESCRIPTION:11]: done

**Pull request merged**